Marvel/Marvel/Clean/Data/Api/MarvelApi.swift

```diff
@@ -10,88 +10,98 @@ import Foundation
 protocol MarvelApiProtocol {
     func fetchCharactersData(completion: @escaping (AllCharactersResponse?, Error?) -> Void)
     func fetchComicsData(completion: @escaping (ComicDataWrrapper?, Error?) -> Void)
-    func getAllCharacters() -> [Character]
 }
 
 class MarvelApi: MarvelApiProtocol {
-    
-    var characters: [Character] = []
-    var comics: [Comic] = []
+
+    static private var shared: MarvelApi?
+    private var characters: [Character] = []
+    private var comics: [Comic] = []
+    private var baseUrl: String
+    private var apiKey: String
+    private var hash: String
+    private var urlsSession: URLSession
+
+    private init(baseUrl: String, apiKey: String, hash: String, urlsSession: URLSession = URLSession.shared ) {
+        self.baseUrl = baseUrl
+        self.urlsSession = urlsSession
+        self.apiKey = apiKey
+        self.hash = hash
+    }
+
+    //TODO : To be improved 
+    static func getInstance() -> MarvelApiProtocol{
+      if let returnShared = shared{
+        return returnShared
+      } else {
+        let newInstance =
+          MarvelApi(baseUrl: ProcessInfo.processInfo.environment["baseUrl"] ?? "",
+                    apiKey: ProcessInfo.processInfo.environment["apiKey"] ?? "",
+                    hash: ProcessInfo.processInfo.environment["hash"] ?? "")
+        shared = newInstance
+        return newInstance
```

Comment:
You can save one line of code by doing:

shared = MarvelApi(baseUrl: ProcessInfo.processInfo.environment["baseUrl"] ?? "",
                    apiKey: ProcessInfo.processInfo.environment["apiKey"] ?? "",
                    hash: ProcessInfo.processInfo.environment["hash"] ?? "")

return shared